### PR TITLE
Session Tracker Update

### LINKED ISSUE #49: Session Tracker and Page
I'm thinking about working on the session tracker next, and wanted to share some ideas I had for it, and hopefully get some of your own, @dm0896665.

Currently we store "sessions" as an aggregation of a player's activity (kills, xp earned, etc.) from login to logout as a record in the DB. I think the general layout of the page should have:
- List of all sessions available to view (page lands on most recent session)
- If the player is online, the current session is available to be viewed and is updated live. Individual events are displayed as they come in.
- non-MVP goal: when a session ends, generate a path on the continent map that shows where the player went (obviously we don't do this for a current/live session, as that would allow people to be tracked).

### PR DESCRIPTION
**Pull request by Bentheburrito**
~~:warning: **Please review, approve, and merge #139 first** :warning:~~ :heavy_check_mark: 

This PR introduces lots of changes and additions to the session page, including (but not limited to):
- Summary stats now update live
- All occurances of character names/IDs should now be click-able in the Event Log (when clicked, will bring you to their general stats page)
- Many new events, such as revives, kill assists, gunner kills, and more, now show up in the Event Log
- Once a session ends, a Kills By Weapon section is shown
- A map of Facilities are now stored in static_data

Closes #49 and #132

### CODE REVIEW

File: lib/sacastats_web/templates/character/template.html.heex [hunk 1]
```diff
@@ -52,7 +52,7 @@
 				<p class="d-inline">Stats last updated on: </p>
 			</div>
 			<div class="col-12 col-md-auto pl-3">
-				<strong><p class="d-inline date-time">
```
Comment (dm0896665) [inline lib/sacastats_web/templates/character/template.html.heex:55]: Why was this removed?

File: lib/sacastats_web/views/character_view.ex [hunk 2]
```diff
@@ -62,6 +62,27 @@ defmodule SacaStatsWeb.CharacterView do
     |> Enum.join(" ")
   end
 
+  def build_kills_by_weapon(assigns, %Session{} = session) do
+    weapon_kill_counts =
+      for %Death{} = death <- session.deaths, reduce: %{} do
+        weapon_kill_counts ->
+          if is_nil(SacaStats.weapons()[death.attacker_weapon_id]) do
```
Comment (dm0896665) [inline lib/sacastats_web/views/character_view.ex:69]: Should probably add a check if killer id = character id

File: lib/sacastats_web/views/character_view.ex [hunk 3]
```diff
@@ -98,23 +138,81 @@ defmodule SacaStatsWeb.CharacterView do
...
+
+    facility_type_text =
+      if facility["facility_type"] in ["Small Outpost", "Large Outpost"] do
+        ""
+      else
+        facility["facility_type"]
+      end
+
+    outfit_captured_text = ""
+    # if def.outfit_id == session.outfit.outfit_id do
```
Comment (dm0896665) [inline lib/sacastats_web/views/character_view.ex:180]: Was this commented out on purpose?

File: lib/sacastats_web/templates/character/template.html.heex (hunk 1 above)
Comment (Bentheburrito) [reply lib/sacastats_web/templates/character/template.html.heex:55]: because `prettify_timestamp/2` adds an HTML element with the `date-time` class, so it's not needed here (and I think it was interfering with the date formatting iirc)

File: lib/sacastats_web/views/character_view.ex (hunk 2 above)
Comment (Bentheburrito) [reply lib/sacastats_web/views/character_view.ex:69]: yes good catch, I'll add that

File: lib/sacastats_web/views/character_view.ex (hunk 3 above)
Comment (Bentheburrito) [reply lib/sacastats_web/views/character_view.ex:180]: Yeah, for the same reason as L150 above (once we start listening to FacilityControl events and do proper member-outfit contributions/tracking, I'll probably move this logic there).

File: lib/sacastats_web/views/character_view.ex (hunk 2 above)
Comment (Bentheburrito) [reply lib/sacastats_web/views/character_view.ex:71]: Just pushed, let me know if it looks good

**Review by dm0896665 — APPROVED** [review]
